src/Analyzers/CSharp/Analyzers/UseLocalFunction/CSharpUseLocalFunctionDiagnosticAnalyzer.cs

```diff
@@ -156,17 +163,34 @@ private void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext, INamedTyp
                     additionalLocations,
                     properties: null));
 
-                var anonymousFunctionStatement = anonymousFunction.GetAncestor<StatementSyntax>();
-                if (anonymousFunctionStatement != null && localDeclaration != anonymousFunctionStatement)
+                if (shouldReportOnAnonymousFunctionStatement)
                 {
                     syntaxContext.ReportDiagnostic(DiagnosticHelper.Create(
                         Descriptor,
-                        anonymousFunctionStatement.GetLocation(),
+                        anonymousFunctionStatement!.GetLocation(),
                         severity,
                         additionalLocations,
                         properties: null));
                 }
             }
+
+            static bool IsInAnalysisSpan(
+                SyntaxNodeAnalysisContext context,
+                LocalDeclarationStatementSyntax localDeclaration,
+                StatementSyntax? anonymousFunctionStatement,
+                bool shouldReportOnAnonymousFunctionStatement)
+            {
+                if (context.ShouldAnalyzeSpan(localDeclaration.Span))
+                    return true;
+
+                if (shouldReportOnAnonymousFunctionStatement
+                    && context.ShouldAnalyzeSpan(anonymousFunctionStatement!.Span))
```

Comment:
are we trying to be far too precise here?  is there some reasonable 'top node' that we can just do all the checks on that will suffice?  like the `localDeclaration`?

Reply:
I am preserving the original semantics of checking the exact node on which we can report a diagnostic.